### PR TITLE
Ensure using correct rmw implementation in tests

### PR DESCRIPTION
**Pull request by dhood**
Connects to ros2/examples#96

### CODE REVIEW

File: test_communication/CMakeLists.txt [hunk 1]
```diff
@@ -123,7 +123,8 @@ if(BUILD_TESTING)
         INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}.py.configured"
       )
 
-      ament_add_nose_test(publisher_subscriber_cpp${test_suffix} "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py" TIMEOUT 15)
+      ament_add_nose_test(publisher_subscriber_cpp${test_suffix} "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py" TIMEOUT 15
+        ENV RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
```
Comment (dirk-thomas) [inline test_communication/CMakeLists.txt]: This test invokes two binaries which might use different RMW impl. therefore a single environment variable can't be used to test.

This needs to be reverted. Same below for the `requester_replier_*` tests.

Comment (dirk-thomas) [inline test_communication/CMakeLists.txt]: I will revert these two changes in 412319ea683385ce259f73f70e7c86636447c2d8 which is part of #119.

Comment (wjwwood) [inline test_communication/CMakeLists.txt]: I guess the `rmw_implementation` needs to be passed down into the `.py` file and then the env be set by `launch`.

Can we just roll that change into the pr that re-enables all the tests that were previously disabled?

Comment (mikaelarguedas) [inline test_communication/CMakeLists.txt]: agreed, all the reenabled tests will fail without that change, so roll that change in https://github.com/ros2/system_tests/pull/119 seems to be the way to go.

Comment (dirk-thomas) [inline test_communication/CMakeLists.txt]: For now I will remove the check in #119. The check can be added afterwards in a separate PR.

Comment (dhood) [inline test_communication/CMakeLists.txt]: @wjwwood did I hear you say you're working on this? otherwise I'll fix it

Comment (wjwwood) [inline test_communication/CMakeLists.txt]: I got it, thanks.

Comment (dhood) [inline test_communication/CMakeLists.txt]: ok thank you!

Comment (wjwwood) [inline test_communication/CMakeLists.txt]: https://github.com/ros2/system_tests/pull/126

**Pull request merged**